backend/control/control.go

```diff
@@ -169,13 +120,130 @@ func TriggerScan(postgresDb *gorm.DB) error {
 		return err
 	}
 	if sr.Status == "FAILED" {
-		return fmt.Errorf("starting cartography job failed")
+		return fmt.Errorf("Scan Status: Failed")
+	}
+	return nil
+}
+
+func CartographyExecutionLoop(postgresDb *gorm.DB, driver neo4j.Driver, ruleDataList []models.RuleData,
```

Comment:
nit: can we just call this ExecutionLoop

Reply:
Didn't got your point?